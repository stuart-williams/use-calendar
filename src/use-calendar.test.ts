import DateFnsAdapter from "@date-io/date-fns";
import { act, renderHook } from "@testing-library/react-hooks";
import locale from "date-fns/locale/en-GB";
import MockDate from "mockdate";

import useCalendar from "./use-calendar";

describe("useCalendar", () => {
  let dateUtils: DateFnsAdapter;

  beforeEach(() => {
    MockDate.set("2000-01-01:09:51");

    dateUtils = new DateFnsAdapter({ locale });
  });

  afterEach(() => {
    MockDate.reset();
  });

  it("should create default calendar", () => {
    const { result } = renderHook(() =>
      useCalendar({
        dateUtils,
      })
    );

    expect(result.current.date).toEqual(new Date("2000-01-01:00:00"));

    expect(result.current.month).toEqual({
      month: "January",
      monthAndYear: "January 2000",
    });

    expect(result.current.weekdays).toEqual([
      { weekday: "Monday", weekdayShort: "Mon" },
      { weekday: "Tuesday", weekdayShort: "Tue" },
      { weekday: "Wednesday", weekdayShort: "Wed" },
      { weekday: "Thursday", weekdayShort: "Thu" },
      { weekday: "Friday", weekdayShort: "Fri" },
      { weekday: "Saturday", weekdayShort: "Sat" },
      { weekday: "Sunday", weekdayShort: "Sun" },
    ]);

    expect(result.current.days).toMatchSnapshot();
  });

  it("should create calendar starting at date", () => {
    const { result } = renderHook(() =>
      useCalendar({
        dateUtils,
        defaultDate: new Date("2050-12-25:09:51"),
      })
    );

    expect(result.current.date).toEqual(new Date("2050-12-25:00:00"));

    expect(result.current.month).toEqual({
      month: "December",
      monthAndYear: "December 2050",
    });

    expect(result.current.days).toMatchSnapshot();
  });

  it("should navigate into the future", () => {
    const { result } = renderHook(() =>
      useCalendar({
        dateUtils,
      })
    );

    act(() => {
      result.current.navigateNext();
    });

    expect(result.current.date).toEqual(new Date("2000-02-01:00:00"));
    expect(result.current.days).toMatchSnapshot();

    act(() => {
      result.current.navigateNext();
    });

    expect(result.current.date).toEqual(new Date("2000-03-01:00:00"));
    expect(result.current.days).toMatchSnapshot();
  });

  it("should navigate into the past", () => {
    const { result } = renderHook(() =>
      useCalendar({
        dateUtils,
      })
    );

    act(() => {
      result.current.navigatePrev();
    });

    expect(result.current.date).toEqual(new Date("1999-12-01:00:00"));
    expect(result.current.days).toMatchSnapshot();

    act(() => {
      result.current.navigatePrev();
    });

    expect(result.current.date).toEqual(new Date("1999-11-01:00:00"));
    expect(result.current.days).toMatchSnapshot();
  });

  it("should navigate to today", () => {
    const { result } = renderHook(() =>
      useCalendar({
        dateUtils,
        defaultDate: new Date("2050-12-25:09:51"),
      })
    );

    expect(result.current.date).toEqual(new Date("2050-12-25:00:00"));

    act(() => {
      result.current.navigateToday();
    });

    expect(result.current.date).toEqual(new Date("2000-01-01:00:00"));
  });

  it("should navigate to date", () => {
    const { result } = renderHook(() =>
      useCalendar({
        dateUtils,
      })
    );

    expect(result.current.date).toEqual(new Date("2000-01-01:00:00"));

    act(() => {
      result.current.navigateTo(new Date("2050-12-25:09:51"));
    });

    expect(result.current.date).toEqual(new Date("2050-12-25:00:00"));
  });
});
