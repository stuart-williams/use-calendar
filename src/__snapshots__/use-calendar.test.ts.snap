// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useCalendar should create default calendar 1`] = `
Array [
  Object {
    "date": "2021-03-28T00:00:00.000Z",
    "dayOfMonth": "28",
    "isOutsideMonth": true,
    "isToday": false,
  },
  Object {
    "date": "2021-03-29T00:00:00.000Z",
    "dayOfMonth": "29",
    "isOutsideMonth": true,
    "isToday": false,
  },
  Object {
    "date": "2021-03-30T00:00:00.000Z",
    "dayOfMonth": "30",
    "isOutsideMonth": true,
    "isToday": false,
  },
  Object {
    "date": "2021-03-31T00:00:00.000Z",
    "dayOfMonth": "31",
    "isOutsideMonth": true,
    "isToday": false,
  },
  Object {
    "date": "2021-04-01T00:00:00.000Z",
    "dayOfMonth": "1",
    "isOutsideMonth": false,
    "isToday": true,
  },
  Object {
    "date": "2021-04-02T00:00:00.000Z",
    "dayOfMonth": "2",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-03T00:00:00.000Z",
    "dayOfMonth": "3",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-04T00:00:00.000Z",
    "dayOfMonth": "4",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-05T00:00:00.000Z",
    "dayOfMonth": "5",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-06T00:00:00.000Z",
    "dayOfMonth": "6",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-07T00:00:00.000Z",
    "dayOfMonth": "7",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-08T00:00:00.000Z",
    "dayOfMonth": "8",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-09T00:00:00.000Z",
    "dayOfMonth": "9",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-10T00:00:00.000Z",
    "dayOfMonth": "10",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-11T00:00:00.000Z",
    "dayOfMonth": "11",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-12T00:00:00.000Z",
    "dayOfMonth": "12",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-13T00:00:00.000Z",
    "dayOfMonth": "13",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-14T00:00:00.000Z",
    "dayOfMonth": "14",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-15T00:00:00.000Z",
    "dayOfMonth": "15",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-16T00:00:00.000Z",
    "dayOfMonth": "16",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-17T00:00:00.000Z",
    "dayOfMonth": "17",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-18T00:00:00.000Z",
    "dayOfMonth": "18",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-19T00:00:00.000Z",
    "dayOfMonth": "19",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-20T00:00:00.000Z",
    "dayOfMonth": "20",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-21T00:00:00.000Z",
    "dayOfMonth": "21",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-22T00:00:00.000Z",
    "dayOfMonth": "22",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-23T00:00:00.000Z",
    "dayOfMonth": "23",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-24T00:00:00.000Z",
    "dayOfMonth": "24",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-25T00:00:00.000Z",
    "dayOfMonth": "25",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-26T00:00:00.000Z",
    "dayOfMonth": "26",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-27T00:00:00.000Z",
    "dayOfMonth": "27",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-28T00:00:00.000Z",
    "dayOfMonth": "28",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-29T00:00:00.000Z",
    "dayOfMonth": "29",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-04-30T00:00:00.000Z",
    "dayOfMonth": "30",
    "isOutsideMonth": false,
    "isToday": false,
  },
  Object {
    "date": "2021-05-01T00:00:00.000Z",
    "dayOfMonth": "1",
    "isOutsideMonth": true,
    "isToday": false,
  },
]
`;
